mergify_engine/clients/http.py

```diff
@@ -64,12 +64,16 @@ class HTTPClientSideError(httpx.HTTPStatusError):
     def message(self) -> str:
         # TODO(sileht): do something with errors and documentation_url when present
         # https://developer.github.com/v3/#client-errors
-        to_cast = (
-            self.response.json()["errors"][0]["message"]
-            if "errors" in self.response.json()
-            else self.response.json()["message"]
-        )
-        return typing.cast(str, to_cast)
+        response = self.response.json()
+        message = response["message"] if "message" in response else "No message found"
+
+        if "errors" in response:
+            if "message" in response["errors"][0]:
+                message = response["errors"][0]["message"]
+            elif isinstance(response["errors"][0], str):
+                message = response["errors"][0]
+
```

Comment:
```suggestion
        if "errors" in response:
            if "message" in response["errors"][0]:
                message = response["errors"][0]["message"]
            elif isinstance(response["errors"][0], str):
                message = response["errors"][0]
        else:
            message = response.get("message", "No message found")
```

Reply:
If we do this we are not catching the follow format: 

`{"message":"Server Error: Sorry, this diff is taking too long to generate.","errors":[{"resource":"PullRequest","field":"diff","code":"not_available"}],"documentation_url":"https://docs.github.com/rest/reference/pulls#list-pull-requests-files"}
`